test/e2e-tests-upgrade.sh

```diff
@@ -16,19 +16,40 @@
 
 # This script calls out to scripts in tektoncd/plumbing to setup a cluster
 # and deploy Tekton Pipelines to it for running upgrading tests. There are
-# two scenarios we need to cover in this script:
+# three scenarios we need to cover in this script:
 
 # Scenario 1: install the previous release, upgrade to the current release, and
 # validate whether the Tekton pipeline works.
 
 # Scenario 2: install the previous release, create the pipelines and tasks, upgrade
 # to the current release, and validate whether the Tekton pipeline works.
 
+# Scenario 3: install the current pipelines, test the current main devel against
+# the integration tests from the previous release(s) to prevent regression. Incompatible 
+# changes made in the current release cycle are expected to fail and will be skipped after
+# failing once.
+
 source $(git rev-parse --show-toplevel)/test/e2e-common.sh
 RELEASES='https://github.com/tektoncd/pipeline/releases/latest'
 PREVIOUS_PIPELINE_VERSION=$(curl -L -s -H 'Accept: application/json' $RELEASES |
   sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
 
+# BACKWARD_INCOMPATIBLE_TESTS_FILES, TESTS and EXAMPLES are manually maintained list of e2e tests that contains
+# known breaking changes made in the current devel cycle. They are updated to prevent the upgrade test job from
+# keeping failing after known breaking changes are made.
+BACKWARD_INCOMPATIBLE_EXAMPLES=(pipelineruns/pipelinerun taskruns/workspace-in-sidecar)
```

Comment:
curious why these tests are exempted? FYI, workspace in sidecar is known to be flaky